src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -710,5 +718,108 @@ protected bool IsAlive
             }
         }
 
+        #region Location extraction & manipulation
+
+        /// <summary>
+        /// Update an existing element's location
+        /// </summary>
+        /// <param name="geometry">New Location Point or Curve</param>
+        public void SetLocation(Geometry geometry)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
+
+            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint))
+            {
+                if (geometry.GetType() == typeof(Autodesk.DesignScript.Geometry.Point))
+                {
+                    Autodesk.DesignScript.Geometry.Point point = (Autodesk.DesignScript.Geometry.Point)geometry;
+                    Autodesk.Revit.DB.LocationPoint pt = (Autodesk.Revit.DB.LocationPoint)this.InternalElement.Location;
+                    pt.Point = point.ToRevitType(true);
+                }
+                else
+                    throw new Exception("Point element required");
+            }
+            else if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationCurve) && geometry is Curve)
+            {
+                if (geometry.GetType() == typeof(Curve))
```

Comment:
What if `geometry` type is derived from Curve? Since you are anyway going to cast it later on, you can better cast the geometry to Curve before entering the if() block.
`var dynamoCurve = geometry as Curve; if(dynamoCurve != null){}`

You can do the same for other checks as well.